### PR TITLE
Fix: cypress `findByTestId` and `percySnapshotElement` do not work together

### PR DESCRIPTION
**Pull request by kerryarchibald**
Fixes https://github.com/vector-im/element-web/issues/25391

## Checklist

-   [ ] Tests written for new code (and old code if feasible)
-   [ ] Linter and other CI checks pass
-   [ ] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-react-sdk/blob/develop/CONTRIBUTING.md))

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

### CODE REVIEW

**Review by weeman1337 — APPROVED** [review]

File: cypress/support/percy.ts [hunk 1]
```diff
@@ -47,6 +47,13 @@ Cypress.Commands.add("percySnapshotElement", { prevSubject: "element" }, (subjec
...
     }
+
+    let selector = subject.selector;
+    // cy.findByTestId sets the selector to `findByTestId(<testId>)`
+    // which is not usable as a scope
+    if (selector.startsWith("findByTestId")) {
+        selector = `[data-testid="${subject.attr("data-testid")}"]`;
+    }
     cy.percySnapshot(name, {
         domTransformation: (documentClone) => scope(documentClone, subject.selector),
```
Comment (t3chguy) [inline cypress/support/percy.ts:58]:
```suggestion
        domTransformation: (documentClone) => scope(documentClone, selector),
```

**Review by t3chguy — APPROVED** [review]

File: cypress/support/percy.ts (hunk 1 above)
Comment (t3chguy) [reply cypress/support/percy.ts:58]: The variable was previously unused

**Comment by t3chguy** [comment]
Running a percy test

**Comment by t3chguy** [comment]
That unfortunately still did not work

**Comment by t3chguy** [comment]
Somehow my earlier change didn't take effect so the test was invalid

**Comment by t3chguy** [comment]
image

Yay

**Pull request merged**